### PR TITLE
[MM-17510] Make `/jira connect` open in a popup for desktop app

### PR DESCRIPTION
**Pull request by mickmister**
#### Summary

This PR makes it so when the `/jira connect` command is run in the desktop app, the redirect opens in a popup, rather than the browser.

This makes the server-side `connect` command useless, as it is now handled client-side. Should the command be outright removed from the server? 

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-17510

### CODE REVIEW

**Comment by mickmister** [comment]
@levb Good thinking. I've confirmed that neither work on mobile (server-generated link and server-generated redirect). The link has the issues with cookies, and the redirect doesn't have any affect on the app because we are not in a browser. There isn't even a handler for `GotoLocation` in the mobile app.

So I think we're good to remove the command server-side.

**Comment by mickmister** [comment]
@levb Also confirmed that the client-side processing works on mobile-browser as well.

**Comment by mickmister** [comment]
@levb I think I'll keep the server-based command there, but just return an error saying that you have to connect on the desktop or webapp, because we know it is coming from the mobile app.

File: server/command.go [hunk 1]
```diff
@@ -96,22 +96,7 @@ func executeConnect(p *Plugin, c *plugin.Context, header *model.CommandArgs, arg
...
-		return p.responsef(header, "You already have a Jira account linked to your Mattermost account. Please use `/jira disconnect` to disconnect.")
-	}
-
-	redirectURL, err := instance.GetUserConnectURL(header.UserId)
-	if err != nil {
-		return p.responsef(header, "Command failed, please contact your system administrator: %v", err)
-	}
-
-	return p.responseRedirect(redirectURL)
+	return p.responsef(header, "Please connect your account using the desktop or browser client.")
```
Comment (mickmister) [inline server/command.go]: I took out the bits checking for an existing instance / connected account because it is irrelevant in this context.

**Comment by levb** [comment]
@mickmister so why keep the server command? RN?

**Comment by mickmister** [comment]
@levb Yes do you think we should instead just get rid of it? It may be confusing to the user to just see the help text. It's not too important, I can get rid of it no problem if that's the consensus.

**Comment by levb** [comment]
> It may be confusing to the user to just see the help text.

Which users will ever see it, if it's always intercepted by the client?

**Comment by levb** [comment]
@aaronrothschild the choice is yours:
(a) get rid of the server-side `/jira connect` altogether. Pros: saves a few lines of code, and some potential confusion over what is implemented where. Cons: will do _nothing_ on RN, displays the help text
(b) the opposite of (a) - keep the server side implementation, effectively for RN only where it will not work then anyway, but with a specific message.

**Comment by mickmister** [comment]
I vote for (a)

**Comment by aaronrothschild** [comment]
I vote for A as well.

File: webapp/src/hooks/hooks.js [hunk 2]
```diff
@@ -23,6 +24,20 @@ export default class Hooks {
             this.store.dispatch(openCreateModalWithoutPost(description, contextArgs.channel_id));
             return Promise.resolve({});
         }
+
+        if (message && (message.startsWith('/jira connect') || message === '/jira connect')) {
```
Comment (levb) [inline webapp/src/hooks/hooks.js]: redundant

Comment (mickmister) [reply webapp/src/hooks/hooks.js]: I will address this in a separate PR once this and `subscribe` are merged

**Comment by levb** [comment]
@mickmister can you merge master, please?

**Pull request merged**